library/bdImage/bdApi/ControllerApi/Post.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+class bdImage_bdApi_ControllerApi_Post extends XFCP_bdImage_bdApi_ControllerApi_Post
+{
+    public function actionGetEditorConfig()
+    {
+        $response = parent::actionGetEditorConfig();
+
+        if ($this->_isFieldIncluded('post_images') && ($response instanceof XenForo_ControllerResponse_View)) {
+            $input = $this->_input->filter(array('post_id' => XenForo_Input::UINT));
+            list($post,,) = $this->_getForumThreadPostHelper()->assertPostValidAndViewable($input['post_id']);
+
+            /** @var bdImage_XenForo_DataWriter_DiscussionMessage_Post $firstPostDw */
+            $firstPostDw = XenForo_DataWriter::create('XenForo_DataWriter_DiscussionMessage_Post');
+            $firstPostDw->setExistingData($post, true);
+
+            $postBodyImage = array();
+            $imageDataList = $firstPostDw->bdImage_extractImage(true);
+            foreach ($imageDataList as $imageData) {
+                $data = json_decode($imageData, true);
```

Comment:
Không xài json_decode nha.


```suggestion
                $data = bdImage_Helper_Data::unpack($imageData);
```